aea/aea_builder.py

```diff
@@ -1336,16 +1338,33 @@ def from_aea_project(
         builder = AEABuilder(with_default_packages=False)
 
         # load agent configuration file
-        configuration_file = aea_project_path / DEFAULT_AEA_CONFIG_FILE
+        configuration_file = cls.get_configuration_file_path(aea_project_path)
+        agent_configuration = cls.loader.load(configuration_file.open())
 
-        loader = ConfigLoader.from_configuration_type(PackageType.AGENT)
-        agent_configuration = loader.load(configuration_file.open())
+        builder.set_from_configuration(
+            agent_configuration, aea_project_path, skip_consistency_check
+        )
+        return builder
+
+    @classmethod
+    def from_config_json(
+        cls, json_data, aea_project_path: PathLike, skip_consistency_check: bool = False
+    ):
+        aea_project_path = Path(aea_project_path)
+        builder = AEABuilder(with_default_packages=False)
+
+        # load agent configuration file
+        agent_configuration = cls.loader._load_agent_config_from_json(json_data)
```

Comment:
The private method should be made public in the loader

Reply:
done